hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyRequest.java

```diff
@@ -452,15 +457,12 @@ protected void getFileEncryptionInfoForMpuKey(KeyArgs keyArgs,
       acquireLock = omMetadataManager.getLock().acquireReadLock(
           BUCKET_LOCK, volumeName, bucketName);
       try {
-
         ResolvedBucket resolvedBucket = ozoneManager.resolveBucketLink(
             Pair.of(keyArgs.getVolumeName(), keyArgs.getBucketName()));
-
-        OmKeyInfo omKeyInfo = omMetadataManager.getOpenKeyTable().get(
-            omMetadataManager.getMultipartKey(resolvedBucket.realVolume(),
-                resolvedBucket.realBucket(), keyArgs.getKeyName(),
-                keyArgs.getMultipartUploadID()));
-
+        OmKeyInfo omKeyInfo =
+            omMetadataManager.getOpenKeyTable(getBucketLayout()).get(
```

Comment:
@aryangupta1998 
It should be `resolvedBucket.realVolume()`  and `resolvedBucket.realBucket()`. Please modify it and is the reason for `TestOzoneAtRestEncryption#testMPUwithLinkBucket`

Reply:
Done